c++/cpp2py/converters/vector.hpp

```diff
@@ -9,23 +9,22 @@
 namespace cpp2py {
 
   template <typename T> static void delete_pycapsule(PyObject *capsule) {
-    auto *ptr = static_cast<std::unique_ptr<T[]> *>(PyCapsule_GetPointer(capsule, "guard"));
+    auto *ptr = static_cast<std::vector<T> *>(PyCapsule_GetPointer(capsule, "guard"));
     delete ptr;
   }
 
   // Convert vector to numpy_proxy, WARNING: Deep Copy
-  template <typename T> numpy_proxy make_numpy_proxy_from_vector(std::vector<T> const &v) {
+  template <typename T> numpy_proxy make_numpy_proxy_from_vector(std::vector<T> v) {
 
-    auto *data_ptr = new std::unique_ptr<T[]>{new T[v.size()]};
-    std::copy(begin(v), end(v), data_ptr->get());
-    auto capsule = PyCapsule_New(data_ptr, "guard", &delete_pycapsule<T>);
+    auto *vec_heap = new std::vector<T>{std::move(v)};
+    auto capsule   = PyCapsule_New(vec_heap, "guard", &delete_pycapsule<T>);
 
     return {1, // rank
             npy_type<std::remove_const_t<T>>,
```

Comment:
here we use remove_const and not decay.
why ?

Reply:
This is copy-cat from the nda / triqs::array implementation.
All the `decay_t` above should probably be `remove_const_t` afterall, as this is what we intend.